src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -1795,6 +1803,78 @@ static IEnumerable<Decorator> GetBicepTemplateDecorators(IFeatureProvider featur
                     })
                     .Build();
 
+                if (featureProvider.WaitAndRetryEnabled)
+                {
+                    yield return new DecoratorBuilder(LanguageConstants.WaitUntilPropertyName)
+                    .WithDescription("Causes the resource deployment to wait until the given condition is satisfied")
+                    .WithRequiredParameter("predicate", OneParamLambda(LanguageConstants.Object, LanguageConstants.Bool), "The predicate applied to the resource.")
+                    .WithRequiredParameter("maxWaitTime", LanguageConstants.String, "Maximum time used to wait until the predicate is true. Please be cautious as max wait time adds to total deployment time. It cannot be a negative value. Use [ISO 8601 duration format](https://en.wikipedia.org/wiki/ISO_8601#Durations).")
+                    .WithFlags(FunctionFlags.ResourceDecorator)
+                    .Build();
+
+                    yield return new DecoratorBuilder(LanguageConstants.RetryOnPropertyName)
+                    .WithDescription("Causes the resource deployment to retry when deployment failed with one of the exceptions listed")
+                    .WithRequiredParameter("exceptionCodes", LanguageConstants.StringArray, "List of exceptions.")
+                    .WithOptionalParameter("retryCount", LanguageConstants.Int, "Maximum number if retries on the exception.")
```

Comment:
You can skip the range check below if you use a constrained integer here.
```suggestion
                    .WithOptionalParameter("retryCount", TypeFactor.CreateIntegerType(minValue: 1), "Maximum number if retries on the exception.")
```

Reply:
Thanks for this suggestion. I am learning the codebase in-depth with these comments. I have made the code changes as suggested.

Reply:
Nice. this is even better than I what I was suggesting.